src/Lib/TableFormatter.php

```diff
@@ -0,0 +1,247 @@
+<?php declare(strict_types=1);
+
+/**
+ * ArrayToTextTable
+ *
+ * Display arrays in terminal
+ *
+ * @author      Mathieu Viossat <mathieu@viossat.fr>
+ * @copyright   Copyright (c) 2015 Mathieu Viossat
+ * @license     http://opensource.org/licenses/MIT
+ * @link        https://github.com/MathieuViossat/arraytotexttable
+ */
+
+namespace Manticoresearch\Buddy\Lib;
+
+class TableFormatter {
+
+	const BORDER_VERTICAL = '|';
+	const BORDER_HORIZONTAL = '-';
+	const BORDER_CORNER = '+';
+
+	/** @var array<int,array<string,string>> $data */
+	protected $data;
+
+	/** @var array<int,string> $keys */
+	protected $keys;
+
+	/** @var array<int> $widths */
+	protected $widths;
+
+	/** @var string $table */
+	protected $table;
+
+	/**
+	 * @param ?array<array<string,mixed>|object> $origData
+	 * @return void
+	 */
+	public function __construct(?array $origData = []) {
+		$this->setData($origData);
+		$this->table = '';
+	}
+
+	public function __toString() {
+		return $this->table;
+	}
+
+	/**
+	 * @param float $startTime
+	 * @param ?array<mixed> $origData
+	 * @param ?int $total
+	 * @return string
+	 */
+	public function getTable(
+		float $startTime,
+		?array $origData = null,
+		?int $total = null,
+		?string $error = null
+	): string {
+		if ($error !== null) {
+			return "ERROR: $error" . PHP_EOL . PHP_EOL;
+		}
+		$table = '';
+		if ($origData !== null) {
+			$this->setData($origData);
+		}
+		$data = $this->prepare();
+		if (!empty($data)) {
+			$borderLine = $this->borderLine();
+			$table .= $borderLine . PHP_EOL;
+			$keysRow = array_combine($this->keys, $this->keys);
+			$table .= implode(PHP_EOL, $this->row($keysRow)) . PHP_EOL;
+			$table .= $borderLine . PHP_EOL;
+			foreach ($data as $row) {
```

Comment:
It's preferred to use `array_map` here
`$table .= implode(PHP_EOL, array_map($this->row(...), $data)) . PHP_EOL;`